aptos-move/block-executor/src/executor.rs

```diff
@@ -527,17 +532,28 @@ where
         block: &TP,
         num_workers: usize,
     ) -> Result<(), PanicOr<ParallelBlockExecutionError>> {
-        let mut block_limit_processor = shared_commit_state.acquire();
+        let block_limit_processor = &mut shared_commit_state.acquire();
+        let mut commit_side_effects = false;
 
-        while let Some((txn_idx, incarnation)) = scheduler.try_commit() {
-            let mut executed_at_commit = false;
-            if !Self::validate_and_commit_delayed_fields(
-                txn_idx,
+        if !Self::validate_and_commit_delayed_fields(txn_idx, versioned_cache, last_input_output)? {
+            // Transaction needs to be re-executed, one final time.
+            commit_side_effects = true;
```

Comment:
Does it cause behavior change?

Reply:
This is used to share some code between two cases: one when there is a re-execution due to delayed fields invalidation - and the output of the txn can then change since the control flow may change. The other case is w.o. re-execution, but having a module publishing in a committed txn. Then, the module becomes visible for all. The shared behavior e.g. in V1 validation index is reduced.